pages/marriage.php

```diff
@@ -148,10 +148,49 @@ function marriage() {
 
     if(isset($_POST['confirm_divorce'])) {
         try {
-            $system->query("UPDATE `users` SET `spouse`='0', `marriage_time`='0' WHERE `user_id`='$player->spouse' LIMIT 1");
-            if(!$system->db_last_affected_rows) {
-                throw new Exception("Error processing divorce!");
+            $result = $system->query("SELECT user_id, user_name, spouse FROM users WHERE user_id ='$player->spouse' LIMIT 1");
+            if (!$system->db_last_num_rows)
+            {
+                throw new Exception("Failed to fetch marriage information");
+            }
+            $current_marriage = $system->db_fetch($result);
+
+            // All data matches, process divorce.
+            if ($current_marriage['spouse'] === $player->user_id) {
+                $system->query("UPDATE `users` SET `spouse`='0', `marriage_time`='0' WHERE `user_id`='$player->spouse' LIMIT 1");
+                if(!$system->db_last_affected_rows) {
+                    throw new Exception("Error processing divorce!");
+                }
             }
+            else
+           {
+               $spouses_spouse_result = $system->query("SELECT user_id, user_name, spouse FROM users WHERE user_id =" . $current_marriage['spouse'] . " LIMIT 1");
+               if (!$system->db_last_num_rows)
+               {
+                   throw new Exception("Failed to fetch spouse information");
+               }
+               $spouses_spouse = $system->db_fetch($spouses_spouse_result);
+               // Players marriage is invalid and spouse has invalid marriage
+               if (!$current_marriage['user_id'] === $spouses_spouse['spouse'])
+               {
+                   $system->query("UPDATE `users` SET `spouse`='0', `marriage_time`='0' WHERE `user_id`='$player->spouse' OR `user_id`=" . $current_marriage['spouse'] . " LIMIT 2");
+                   if(!$system->db_last_affected_rows) {
+                       throw new Exception("Error processing divorce!");
+                   }
+               }
+               $spouse_player_result = $system->query("SELECT user_id, user_name, spouse FROM users WHERE spouse ='$player->user_id' LIMIT 1");
+
+               // Another player has an unexpected marriage with player
+               if ($system->db_last_num_rows)
+               {
+                   $spouse_player = $system->db_fetch($spouse_player_result);
+
+                   $system->query("UPDATE `users` SET `spouse`='0', `marriage_time`='0' WHERE `user_id`=". $spouse_player['user_id'] ." LIMIT 1");
+                   if(!$system->db_last_affected_rows) {
+                       throw new Exception("Error processing divorce!");
+                   }
+               }
+           }
```

Comment:
I think we can drop this whole logic, if the spouse is not marked as married to the current player we can just clear the current player's spouse field, and the other person can sort it out on their end